deps/rabbitmq_cli/rabbitmqctl.bzl

```diff
@@ -180,15 +207,144 @@ rabbitmqctl_private = rule(
     toolchains = [
         "//bazel/elixir:toolchain_type",
     ],
-    provides = [ErlangAppInfo],
+    provides = [ElixirAppInfo],
     executable = True,
 )
 
-def rabbitmqctl(**kwargs):
+def _elixir_app_to_erlang_app(ctx):
+    app_consolidated = ctx.attr.elixir_app[ElixirAppInfo].consolidated
+    app_ebin = ctx.attr.elixir_app[ElixirAppInfo].beam
+
+    elixir_ebin = ctx.attr.elixir_as_app[ErlangAppInfo].beam[0].path
+
+    ebin = ctx.actions.declare_directory(path_join(ctx.label.name, "ebin"))
+
+    if ctx.attr.mode == "elixir":
+        ctx.actions.run_shell(
+            inputs = ctx.files.elixir_as_app + ctx.files.elixir_app,
+            outputs = [ebin],
+            command = """\
+set -euo pipefail
+
+cp {elixir_ebin}/* {ebin}
+
+for beam in {app_consolidated}/*; do
+    find {ebin} -name "$(basename $beam)" -exec cp -f $beam {ebin} \\;
+done
+""".format(
+                elixir_ebin = elixir_ebin,
+                app_consolidated = app_consolidated.path,
+                ebin = ebin.path,
+            ),
+        )
+
+        lib_info = ctx.attr.elixir_as_app[ErlangAppInfo]
+        return [
+            DefaultInfo(files = depset([ebin])),
+            ErlangAppInfo(
+                app_name = "elixir",
+                include = lib_info.include,
+                beam = [ebin],
+                priv = lib_info.priv,
+                deps = lib_info.deps,
+            ),
+        ]
+    elif ctx.attr.mode == "app":
+        ctx.actions.run_shell(
+            inputs = ctx.files.elixir_as_app + ctx.files.elixir_app,
+            outputs = [ebin],
+            command = """\
+set -euo pipefail
+
+cp {app_ebin}/* {ebin}
+cp -f {app_consolidated}/* {ebin}
+
+for beam in {elixir_ebin}/*; do
+    find {ebin} -name "$(basename $beam)" -delete
+done
+""".format(
+                elixir_ebin = elixir_ebin,
```

Comment:
I think every substituted bit should be shell-quoted, so it won't break apart with all the strange names.

Reply:
I was thinking about something like https://github.com/bazelbuild/bazel-skylib/blob/main/docs/shell_doc.md#shellquote
In case if someone decides to have a stupidly named file or folder like (literally) `$HOME`

Reply:
In practice, we probably don't have to worry that much here. The files all are either elixir modules, or need to have passed through mix, and bazel would have probably have already failed if the project itself exists in strange dir.
However, I didn't know skylib contained such a helper, so I'll make another pass at it, mostly due to my own curiosity.